src/lib/config.ts

```diff
@@ -1052,13 +1064,13 @@ function setDeepProperty(obj, value, target) {
 export function mergeDeep(o1, o2) {
```

Comment:
It feels a bit naughty to have mergeDeep also remove all empty strings. It was the easiest way of making sure it worked everywhere.

It might make more sense to rename mergeDeep everywhere to e.g. `mergeDeepAndCull`, which simply calls `removeEmpty(mergeDeep(`.